### ISSUE TITLE
permit/deny openvswitch_t type in selinux should run only when selinux is enabled

### ISSUE THREAD

**Issue by vvb** [contiv/ansible#173]
permit/deny openvswitch_t type in selinux should run only when selinux is enabled

currently I see the below failure

```
Apr 21 11:58:47 ucsb-blade1 clusterm: level=info msg="TASK [contiv_network : permit openvswitch_t type in selinux] *******************"
Apr 21 11:58:51 ucsb-blade1 clusterm: level=info msg="fatal: [ucsb-blade3-FLM19145KXV]: FAILED! => {\"changed\": true, \"cmd\": \"semanage permissive -a openvswitch_t\", \"delta\": \"0:00:04.046924\", \"end\": \"2016-04-21 11:58:51.721902\", \"failed\": true, \"rc\": 1, \"start\": \"2016-04-21 11:58:47.674978\", \"stderr\": \"libsepol.link_modules: Tried to link in a non-MLS module with an MLS base. (No such file or directory).\\nlibsemanage.semanage_link_sandbox: Link packages failed (No such file or directory).\\nValueError: Could not commit semanage transaction\", \"stdout\": \"\", \"stdout_lines\": [], \"warnings\": []}"
```

when i execute is manually on the shell,

```
[admin@ucsb-blade2 ~]$ semanage permissive -a openvswitch_t
ValueError: SELinux policy is not managed or store cannot be accessed.
[admin@ucsb-blade2 ~]$
```

**Comment by mapuri** [comment]
@vvb if it's easier to verify, can you verify that above issue goes away if you enable selinux on the box? Somehow the ansible's shell error doesn't match the error that you hit with manual command run. May be there is more, like we may need to set the ansible shell as well?

**Comment by vvb** [comment]
@mapuri will update on this later today. I had modified my setup and ran into some issues

**Comment by gaurav-dalvi** [comment]
@mapuri and @vvb :
I don't see this problem. I have two nodes (centos 7) with swarm setup.

TASK [contiv_network : permit openvswitch_t type in selinux] *******************
changed: [node2]
changed: [node1]

This is when I have selinux is disabled on both nodes:
admin@netmaster ansible]$ sestatus
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   permissive                <------------------ Disabled
Mode from config file:          enforcing
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Max kernel policy version:      28

@vvb: 
You need to do sudo for this command : semanage permissive -a openvswitch_t

[admin@localhost ~]$ semanage permissive -a openvswitch_t
ValueError: SELinux policy is not managed or store cannot be accessed.
[admin@localhost ~]$ sudo semanage permissive -a openvswitch_t
[admin@localhost ~]$

**Comment by abhi** [comment]
@gaurav-dalvi @vvb I encountered this issue with folks trying to bring up contiv on ubuntu boxes. Since SELinux is not enabled in ubuntu (infact not installed and i couldnt enable selinux after installing on ubuntu) we might need to check if selinux is installed and if installed then if it is enabled. Had a short discussion with @mapuri as well . May be that will solve your issue as well @vvb ?

**Comment by gaurav-dalvi** [comment]
@abhinandanpb : Yes. I had discussion with @mapuri  about this. I will put a fix tomorrow for this.

**Comment by vvb** [comment]
@abhinandanpb @gaurav-dalvi yes, we should add the check to see if selinux is installed.

**Comment by mapuri** [comment]
@gaurav-dalvi 

thanks for digging into this.

> admin@netmaster ansible]$ sestatus
> SELinux status: enabled

Doesn't this mean selinux is enabled? I think the status line that you point out implies that the selinux policy is permissive. While you are at it do you mind trying with selinux status disabled? I think it will need [reboot of the vm](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Security-Enhanced_Linux/sect-Security-Enhanced_Linux-Enabling_and_Disabling_SELinux-Disabling_SELinux.html)

**Comment by gaurav-dalvi** [comment]
@mapuri :  I see what are you saying now.
We need to fix this.

1: By default Selinux is in enforced mode when enabled.
2: When we do sudo setenfroce 0 : We basically set Selinux to permissive mode. But Selinux is still enabled.

3: When you edit /etc/selinux/config 

 This file controls the state of SELinux on the system.
 SELINUX= can take one of these three values:
     enforcing - SELinux security policy is enforced.
     permissive - SELinux prints warnings instead of enforcing.
     disabled - No SELinux policy is loaded.
SELINUX=disabled <--------------- THIS means its disabled
 SELINUXTYPE= can take one of three two values:
     targeted - Targeted processes are protected,
     minimum - Modification of targeted policy. Only selected processes are protected. 
     mls - Multi Level Security protection.
SELINUXTYPE=targeted 

After this our net_demo_installer script fails witch this error 

TASK [contiv_network : set aci mode] *******************************************
fatal: [node2]: FAILED! => {"changed": true, "cmd": "contivctl net global set --fabric-mode aci", "delta": "0:00:00.008357", "end": "2016-05-05 13:00:03.603689", "failed": true, "rc": 3, "start": "2016-05-05 13:00:03.595332", "stderr": "time=\"2016-05-05T13:00:03-04:00\" level=error msg=\"Post http://netmaster:9999/api/globals/global/: dial tcp 10.193.231.123:9999: getsockopt: no route to host\" ", "stdout": "", "stdout_lines": [], "warnings": []}
    to retry, use: --limit @./ansible/site.retry

**Comment by vvb** [comment]
@mapuri @gaurav-dalvi so, yes I have verified that this issue occurs only when selinux is disabled. 

@mapuri I re-created it again, and even though the error message doesn't clearly say that it is because of selinux being disable, that is what I am noticing. This error goes away after selinux is enabled.

```
May  5 19:40:13 ucsb-blade1 clusterm: level=info msg="TASK [contiv_network : permit openvswitch_t type in selinux] *******************"
May  5 19:40:17 ucsb-blade1 clusterm: level=info msg="fatal: [ucsb-osd1-FCH1951V212]: FAILED! => {\"changed\": true, \"cmd\": \"semanage permissive -a openvswitch_t\", \"delta\": \"0:00:03.533906\", \"end\": \"2016-05-05 16:40:17.604949\", \"failed\": true, \"rc\": 1, \"start\": \"2016-05-05 16:40:14.071043\", \"stderr\": \"libsepol.link_modules: Tried to link in a non-MLS module with an MLS base. (No such file or directory).\\nlibsemanage.semanage_link_sandbox: Link packages failed (No such file or directory).\\nValueError: Could not commit semanage transaction\", \"stdout\": \"\", \"stdout_lines\": [], \"warnings\": []}"
```

**Comment by gaurav-dalvi** [comment]
@vvb : Excellent. Could you please try this fix in your local setup and check whether it works even if selinux is disabled.

https://github.com/contiv/ansible/pull/187

**Comment by mapuri** [comment]
fixed by #187
